### PR TITLE
[TypeDeclaration] Add TypedPropertyFromJMSSerializerAttributeTypeRector

### LINKED ISSUE #8652: [TypeDeclaration] Add typed property based on serialize attribute
This rule is simple, the JMS annotations state exact type, that can be added to the property.
If on PHH 8.0+ (due to attributes, lets skip annotations to keep rule simple and more reliable), we can add known typed property:

```diff
 [JMS\Type('string')]
-private $name;
+private ?string $name = null;
```

### PR DESCRIPTION
**Pull request by samsonasik**
Closes https://github.com/rectorphp/rector/issues/8652

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

File: rules-tests/TypeDeclaration/Rector/Class_/TypedPropertyFromJMSSerializerAttributeTypeRector/Fixture/skip_typed_assigned.php.inc [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\Class_\TypedPropertyFromJMSSerializerAttributeTypeRector\Fixture;
+
+final class SkipTypedAssigned
+{
+    #[\JMS\Serializer\Annotation\Type('string')]
+    private $name;
+
+    public function setName(string $name): void
```
Comment (samsonasik) [inline rules-tests/TypeDeclaration/Rector/Class_/TypedPropertyFromJMSSerializerAttributeTypeRector/Fixture/skip_typed_assigned.php.inc:10]: assign typed already covered by other rule

https://getrector.com/demo/96ca5ad6-dd87-406e-88bf-6e073ed17e32

File: rules-tests/TypeDeclaration/Rector/Class_/TypedPropertyFromJMSSerializerAttributeTypeRector/Fixture/skip_with_default.php.inc [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\Class_\TypedPropertyFromJMSSerializerAttributeTypeRector\Fixture;
+
+final class SkipWithDefault
+{
+    #[\JMS\Serializer\Annotation\Type('string')]
+    private $name = 'foo';
```
Comment (samsonasik) [inline rules-tests/TypeDeclaration/Rector/Class_/TypedPropertyFromJMSSerializerAttributeTypeRector/Fixture/skip_with_default.php.inc:8]: With default already covered by other rule https://getrector.com/demo/2479ea75-5faa-45b5-8b62-7738cc69d1ec

**Comment by TomasVotruba** [comment]
Looks great :+1:  Thanks for skipping the cases covered by other rules, so we have indepedent rules

**Pull request merged**